weave/cross_thread_com/channels_mpsc_unbounded_batch.nim

```diff
@@ -133,8 +133,11 @@ proc tryRecv*[T](chan: var ChannelMpscUnboundedBatch[T], dst: var T): bool =
   ## This can fail spuriously on the last element if producer
   ## enqueues a new element while the consumer was dequeing it
 
-  let first = cast[T](chan.front.next.load(moRelaxed))
+  let first = cast[T](chan.front.next.load(moAcquire))
```

Comment:
This moRelaxed triggered the race detector every single time:

![image](https://user-images.githubusercontent.com/22738317/81472433-5d36a280-91f8-11ea-804b-6519f5f5e036.png)

The rest of the changes were not checked